src/passes/SimplifyLocals.cpp

```diff
@@ -998,6 +998,11 @@ struct SimplifyLocals
     struct EquivalentOptimizer
       : public LinearExecutionWalker<EquivalentOptimizer> {
 
+      // It is ok to look at adjacent blocks together, as if a later part of a
+      // block is not reached that is fine - changes we make there would not be
+      // reached in that case.
+      bool connectAdjacentBlocks = true;
```

Comment:
Are the cases where we need this the same as where we need `bool ignoreBranchesOutsideOfFunc = true;` for `cfg-traversal.h`?

Reply:
I think it's a little different. I was actually thinking of saying in a comment "this is like ignoreBranchesOutsideOfFunc" but it seemed like it could be more confusing than helpful, though I'm not sure...

The main difference is there we ignore branches outside of the function. We never ignore a branch inside it. So we use that in cases where we just care about state in locals, for example, that goes away when we branch out of the function. We can use that for local coalescing for that reason. But we couldn't use this PR for local coalescing since we need to be aware of every branch inside the function, or else we could miss some interferences.

Reply:
Isn't this PR also about excluding branches out of a function? Ah, I guess it is branch within a function in case of `if`..?

Reply:
Yes, exactly, an if would be inside. I think I can also handle `br_if` etc. but I'll leave that for later.